lib/wallet/src/Cardano/Wallet/Network.hs

```diff
@@ -160,7 +160,8 @@ data NetworkLayer m block = NetworkLayer
         -> m StakePoolsSummary
 
     , getCachedRewardAccountBalance
-        :: RewardAccount
+        :: Either RewardAccount RewardAccount
```

Comment:
Is the distinction between `RewardAccount` and `RewardAccount` really needed, or could we not get away with just `RewardAccount`?

Reply:
yes, one is key hashes based and another is script hashes based. And this information is need when transforming from node reward account. See how this point is used in `Cardano.Wallet.Shelley.Network.Node` module

Reply:
Ah, so I take it really is a binary tag for script vs keyHash object we need for the query, and is not already in the `RewardAccount` bytes...